### DISCUSSION TITLE
Context method to send request to self.

### DISCUSSION THREAD

**Discussion by nicksrandall** [honojs/hono#1675] [Ideas]
Sometimes in a view route, I want to call an another route handler to get data. It would be awesome if Hono has this built in. 

Runtimes like Cloudflare workers do not allow workers to "fetch" themselves over the network so this would be a nice work around that would allow hono to skip the network entirely and just call itself.  

## Proposed API
```jsx
const app = new Hono():

app.get("/api/data", async c => {
  // get some data
  return c.json(data);
});

app.get("/view", async c => {
  const resp = await c.fetch("/api/data", c.req.raw); // follows `Fetch` API
  const data = await resp.json();
  return c.render(<SomeComponent data={data} />);
});
```

I'd be happy to send a PR if maintainers are open to this change.

**Comment by nicksrandall** [comment]
I went ahead and submitted a PR: https://github.com/honojs/hono/pull/1676

**Comment by yusukebe** [comment]
Hi @nicksrandall,

First, please create an issue as a feature request for this. After it's discussed there, if we agree to add the feature, then you can create the PR.

**Comment by nicksrandall** [comment]
Thanks for responding @yusukebe . Here is the issue: #1677

**Reply by yusukebe** [reply]
Thanks!
